test/fixtures/resource-limits.js

```diff
@@ -0,0 +1,24 @@
+'use strict';
+
+const { getHeapSpaceStatistics } = require('v8');
+const assert = require('assert');
+
+module.exports = () => {
+  const array = [];
+  while (true) {
+    let seenSpaces = 0;
+    for (const { space_name } of getHeapSpaceStatistics()) {
+      if (space_name === 'new_space') {
+        seenSpaces++;
+      } else if (space_name === 'old_space') {
+        seenSpaces++;
+      } else if (space_name === 'code_space') {
+        seenSpaces++;
+      }
+    }
+    assert.strictEqual(seenSpaces, 3);
+
+    for (let i = 0; i < 100; i++)
+      array.push([array]);
```

Comment:
```suggestion
    array.push([array]);
```

Asserting that the heap spaces are there makes sense for Node.js core but is rather irrelevant here, tbh